hooks/environment

```diff
@@ -12,6 +12,25 @@ check_command() {
   fi
 }
 
+# In `buildkite-agent` 3.66.0
+# (https://github.com/buildkite/agent/releases/tag/v3.66.0) the ability to
+# redact secrets (https://buildkite.com/docs/agent/v3/cli-redactor) was added.
+# However, because we may not always run that version of `buildkite-agent`, we
+# should conditionally enable it
+check_buildkite_version() {
+  local version_output
+  version_output=$(buildkite-agent --version)
+  local version
+  version=$(echo "$version_output" | grep -oP '\d+\.\d+\.\d+')
+  local min_version="3.66.0"
+  if [ "$(printf '%s\n' "$min_version" "$version" | sort -V | head -n1)" = "$min_version" ]; then
```

Comment:
Also, we should use `[[` when in Bash, as it is better (don't have a reference offhand)
```suggestion
  if [[ "$(printf '%s\n' "$min_version" "$version" | sort -V | head -n1)" = "$min_version" ]]; then
```

Reply:
In similar cases above we use single `[` (https://github.com/elastic/vault-secrets-buildkite-plugin/blob/main/hooks/environment#L63) so I think we should keep the style for consistency. We can consider changing it with a different PR.

Reply:
This is net-new code, which we should be moving towards better practices, especially if we've introduced new code.

I appreciate that refactoring all existing cases may not be reasonable in a given PR, but in this case, migrating the _newly written code_ is practical

We shouldn't be striving for "everything consistently not-great" and then refactoring it after-the-fact - (generally) our codebases are too large to have everything done in a single, consistent way, so we need to do it bit-by-bit

Reply:
Ok, since it is mainly stylish and the difference is not that extensive I can agree with that.

Reply:
changed in 2cab9997094d5f493b5fd50e2ac765f66a5e37de